### PR TITLE
Testsign command

### PR DESCRIPTION
**Pull request by HarrisonPearl**
Implements test sign command.

Command connects client to local signatory over port 3000, and provides a url to be signed. The signatory responds with a signature over the url, generated using the signatory's private key and the public key of the invoked party. 

The Invoked party must have a published public key for the signing to succeed. This command will fail on its first run, as the signatory will not yet have stored a public key for the invoked party. Failure will result in the public key being looked up and stored, and subsequent signing attempts will succeed.  

Example syntax: go run . testsign --url "https://example.com" 
** a signatory must be running locally at port 3000 before this command is run

### CODE REVIEW

File: cmd/testsign.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	"github.com/IABTechLab/adscert/pkg/adscert/signatory"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/protobuf/encoding/prototext"
+)
+
+// testsignCmd represents the test sign command
+var (
+	testsignParams = &testsignParameters{url: "google.com"}
```
Comment (cmlight) [inline cmd/testsign.go:34]: you should be able to omit the default value for the URL here.

File: cmd/testsign.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	conn, err := grpc.Dial(testsignParams.serverAddress, opts...)
+	if err != nil {
+		logger.Fatalf("Failed to dial: %v", err)
+	}
+	defer conn.Close()
+
+	// Optional: performs a health check against the server before actually
+	// trying to invoke the signatory service.
+
+	performOptionalHealthCheckRPC(conn)
```
Comment (cmlight) [inline cmd/testsign.go:77]: I think it would be OK to remove this health check code from this command.

File: cmd/testsign.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	// In most circumstances a signatureResponse will be returned which includes
+	// detals about the successful or failed signature attempt.
+	if signatureResponse != nil {
+		logger.Infof("signature response:\n%s", prototext.Format(signatureResponse))
+	} else {
+		logger.Warningf("signature response is missing")
+	}
+}
+
+func performOptionalHealthCheckRPC(conn *grpc.ClientConn) {
```
Comment (cmlight) [inline cmd/testsign.go:110]: you can also remove this function

File: cmd/testsign.go (hunk 1 above)
Comment (HarrisonPearl) [reply cmd/testsign.go]: done!

File: cmd/testsign.go (hunk 2 above)
Comment (HarrisonPearl) [reply cmd/testsign.go]: done!

File: cmd/testsign.go (hunk 3 above)
Comment (HarrisonPearl) [reply cmd/testsign.go]: done!

**Review by cmlight — COMMENTED** [review]
please also check the presubmit build checks since there is a go.sum update that is needed.

**Review by cmlight — APPROVED** [review]

**Pull request merged**